### ISSUE TITLE
"Setting the OCaml Compilation Flags Globally" example is wrong

### ISSUE THREAD

**Issue by gahr** [ocaml/dune#5435]
## Expected Behavior

I can switch between dev and release builds following the docs.

## Actual Behavior

The release config triggers this compiler error:

```
% dune build --profile release
      ocamlc src/sub/.main.eobjs/byte/dune__exe__Main.{cmi,cmo,cmt} (exit 2)
(cd _build/default && /usr/home/gahr/.opam/default/bin/ocamlc.opt -w -40 -O3 -g -bin-annot -I src/sub/.main.eobjs/byte -no-alias-deps -o src/sub/.main.eobjs/byte/dune__exe__Main.cmo -c -impl src/sub/main.ml)
/usr/home/gahr/.opam/default/bin/ocamlc.opt: unknown option '-O3'.
Usage: ocamlc <options> <files>
Try 'ocamlc --help' for more information.
```

## Reproduction

1. configure your top-level dune file as per [this example](https://github.com/ocaml/dune/blob/main/doc/quick-start.rst#setting-the-ocaml-compilation-flags-globally).
2. run `dune build --profile release`

## Specifications

- Version of `dune` (output of `dune --version`): 2.9.3
- Version of `ocaml` (output of `ocamlc --version`) 4.13.1
- Operating system (distribution and version): FreeBSD 13.0-RELEASE-p4 amd64

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag): https://gist.github.com/gahr/da40bac0569a629b7fe9578ce1e3b2e6

**Comment by jberdine** [comment]
In the gist the `dune` file contains `(flags (:standard -O3))` but the example's use of `ocamlopt_flags` instead of `flags` is crucial since `ocamlc` does not accept `-O3`.

**Comment by Butanium** [comment]
On the version I saw there was no `ocamlopt_flags`, in old version of the file we can see this : 
![image](https://user-images.githubusercontent.com/55806347/156054166-bc2b4487-7f35-4526-b199-18d18a1950f1.png)
Maybe they just actualised the website to the last commits

**Comment by Butanium** [comment]
Actually, I just found the website I was directed to : https://jbuilder.readthedocs.io/en/latest/quick-start.html#setting-the-ocaml-compilation-flags-globally

which doesn't seem up to date

**Comment by gahr** [comment]
> In the gist the `dune` file contains `(flags (:standard -O3))` but the example's use of `ocamlopt_flags` instead of `flags` is crucial since `ocamlc` does not accept `-O3`.

You're right, this was fixed in #4377, and dune.readthedocs.io is correct.
